src/NHibernate/Mapping/Index.cs

```diff
@@ -68,7 +68,8 @@ public static string BuildSqlDropIndexString(Dialect.Dialect dialect, Table tabl
 		/// </returns>
 		public string SqlCreateString(Dialect.Dialect dialect, IMapping p, string defaultCatalog, string defaultSchema)
 		{
-			return BuildSqlCreateIndexString(dialect, Name, Table, ColumnIterator, false, defaultCatalog, defaultSchema);
+			var indexName = IsInherited ? Table.Name + Name : Name;
```

Comment:
Possible improvement to avoid changing index name for dialects where same index name is allowed for different tables.
According to my tests dialects that require unique index name in database:
Firebird
Oracle
SQLite
PostgreSQL

Support same index name:
SQL Server
SQL Server CE
MySql
SAP SQL Anywhere